src/components/QuestionListData.tsx

```diff
@@ -0,0 +1,130 @@
+  displayableQuestions: InterviewQuestionModel[];
+  addableQuestions: InterviewQuestionModel[];
+  allQuestions: InterviewQuestionModel[];
+  addToRemoveDrawerCallback: (question: InterviewQuestionModel) => void;
+  addToAddDrawerCallback: (question: InterviewQuestionModel) => void;
+}): JSX.Element {
+  const searchInput = useRef<Input>();
+
+  const handleSearch = (
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    confirm({ closeDropdown: true });
+  };
+
+  const handleReset = (
+    clearFilters: () => void,
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    clearFilters();
+    confirm({ closeDropdown: true });
+  };
+
+  const tableColumns = (
+    data: InterviewQuestionModel[]
+  ): ColumnsType<InterviewQuestionModel> => {
+    return [
+      {
+        ...createColumnSearchProps(
+          'title',
+          searchInput,
+          handleSearch,
+          handleReset
+        ),
+      },
+      {
+        ...createColumnFilterProps(
+          'category',
+          data
+            .map((value) => value.category!.toString())
+            .filter((value) => value !== undefined)!
+        ),
+      },
+      {
+        ...createColumnSearchProps(
+          'content',
+          searchInput,
+          handleSearch,
+          handleReset
+        ),
+      },
+      {
+        ...createColumnFilterProps(
+          'difficulty',
+          data
+            .map((value) => value.difficulty!.toString())
+            .filter((value) => value !== undefined)!
+        ),
+        sorter: (a: InterviewQuestionModel, b: InterviewQuestionModel) =>
+          a.difficulty != null && b.difficulty != null
+            ? a.difficulty - b.difficulty
+            : 1,
+        render: (value: number) => <Rate disabled value={value} />,
+      },
+    ];
+  };
+
+  if (list?.interviewQuestions) {
+    return isBeingEdited ? (
+      <div className={styles.questionListsData}>
+        <Space direction="vertical">
+          <TableWrapper
+            dataSource={displayableQuestions}
+            columns={tableColumns(list?.interviewQuestions)}
+            customAction={{
+              buttonText: 'Remove',
+              actionCallback: addToRemoveDrawerCallback,
+            }}
+            customTitle="Questions added to list"
+          />
+          <TableWrapper
+            dataSource={addableQuestions}
+            columns={tableColumns(allQuestions)}
+            customAction={{
+              buttonText: 'Add',
+              actionCallback: addToAddDrawerCallback,
+            }}
+            customTitle="Questions you can add to list"
+          />
+        </Space>
+      </div>
+    ) : (
+      <div className={styles.questionListsData}>
+        <TableWrapper
+          dataSource={displayableQuestions}
+          columns={tableColumns(list.interviewQuestions)}
+        />
+      </div>
+    );
+  }
+  return (
+    <div className={styles.questionListsData}>
```

Comment:
I would switch condition and keep rendering of tables at the bottom of function.